### PR TITLE
refactored tests

### PR DESCRIPTION
**Pull request by orthagonal**
refactored tests (one file containing server/network tests and the other
testing various packet formats)
added additional test cases
exposed makeSlackPayload as a server method

### CODE REVIEW

File: test/package-test.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packetString = server.makeSlackPayload(['test'], 'a string');
+    code.expect(typeof packetString).to.equal('string');
+    const packet = JSON.parse(packetString);
+    code.expect(packet.attachments.length).to.equal(1);
```
Comment (jgallen23) [inline test/package-test.js]: probably easier to do a deep equal and just put the whole attachment in there

File: test/package-test.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    code.expect(packet.attachments[0].fields.length).to.equal(1);
+    code.expect(packet.attachments[0].title).to.equal('a string');
+    code.expect(packet.attachments[0].fields[0].title).to.equal('Tags');
+    done();
+  });
+});
+lab.test('lets you post an object as the message', (done) => {
+  server = new Hapi.Server({});
```
Comment (jgallen23) [inline test/package-test.js:48]: you could move this into setup. so each test is just calling server.makeSlackPayload

File: test/package-test.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+let server;
+
+lab.afterEach((done) => {
+  server.stop(() => {
+    done();
+  });
+});
+lab.test('converts a basic message passed as string ', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
```
Comment (jgallen23) [inline test/package-test.js]: is this required? would be great to have potential port conflicts when spinning up tests

File: test/package-test.js (hunk 2 above)
Comment (jgallen23) [inline test/package-test.js:48]: nevermind, I see why you did that.

File: test/package-test.js (hunk 1 above)
Comment (jgallen23) [inline test/package-test.js]: lets switch everything to deep equal so we are making sure there aren't any other things getting added to payload that we weren't expecting

File: test/package-test.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+});
+lab.test('lets you post an object with a special "message" field', (done) => {
+  server = new Hapi.Server({});
+  server.connection({ port: 8080 });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload([], {
+      message: 'this is the message that was pulled out of the object below',
```
Comment (jgallen23) [inline test/package-test.js:147]: need to add another test for `url` too.  will set the title_link field

File: test/package-test.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    options: {
+      username: 'Jared'
+    }
+  }, () => {
+    const packet = JSON.parse(server.makeSlackPayload([], 'a string'));
+    code.expect(packet.username).to.equal('Jared');
+    done();
+  });
+});
+lab.test('will put a link in the title', (done) => {
```
Comment (jgallen23) [inline test/package-test.js]: I think this is mis-labeled

File: test/package-test.js (hunk 4 above)
Comment (jgallen23) [inline test/package-test.js:147]: need a test to test the object if it doesn't have a message field

Comment (jgallen23) [inline test/package-test.js:147]: need a test to test adding the fallback field (will get covered if you do a deep equal on the object)

**Comment by jgallen23** [comment]
can we turn on code coverage? would it be at 100%?

File: test/package-test.js (hunk 4 above)
Comment (orthagonal) [inline test/package-test.js:147]: Can you clarify?  I don't think we have a fallback option currently.

File: test/package-test.js [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+  server = new Hapi.Server({});
+  server.connection({ });
+  server.register({
+    register: hapiSlack,
+    options: {}
+  }, () => {
+    const packetString = server.makeSlackPayload(['test'], 'a string');
+    code.expect(typeof packetString).to.equal('string');
+    const packet = JSON.parse(packetString);
+    code.expect(_.isEqual(packet, expectedPacket)).to.equal(true);
```
Comment (jgallen23) [inline test/package-test.js]: https://github.com/hapijs/code/blob/master/API.md#equalvalue-options

File: test/package-test.js [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+  });
+});
+lab.test('lets you post an object as the message', (done) => {
+  const expectedPacket = {
+    attachments: [{
+      text: '``` {\n  "data": "this is an object"\n} ```',
+      mrkdwn_in: ['text'],
+      fields: [{
+        title: 'Tags',
+        value: ''
```
Comment (jgallen23) [inline test/package-test.js]: we should skip the tags field if no tags are passed

File: test/package-test.js (hunk 4 above)
Comment (jgallen23) [inline test/package-test.js:147]: make sure you merge in from master

File: test/package-test.js (hunk 6 above)
Comment (orthagonal) [inline test/package-test.js]: Nice, I didn't know code had this built in already.

**Pull request merged**